### PR TITLE
Add functionality for unarchiving channel

### PR DESCRIPTION
**Pull request by rehanjaved237**

### CODE REVIEW

**Comment by stakater-user** [comment]
@rehanjaved237 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/8818fe39d15baeae79768fdcc66667a61c13d25e/checks) has Failed!

**Comment by stakater-user** [comment]
@rehanjaved237 Yikes! You better fix it before anyone else finds out! [Build](https://github.com/stakater/slack-operator/commit/458aa6cfd8d48f2ad544163be6cf54ab83fcbb1f/checks) has Failed!

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-32-4300c044`

File: controllers/channel_controller.go [hunk 1]
```diff
@@ -100,9 +101,26 @@ func (r *ChannelReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		log.Info("Creating new channel", "name", name)
 
 		channelID, err := r.SlackService.CreateChannel(name, isPrivate)
-
 		if err != nil {
-			return reconcilerUtil.ManageError(r.Client, channel, err, false)
+			if err.Error() == "name_taken" && os.Getenv("ENABLE_UNARCHIVING_CHANNEL") == "true" {
```
Comment (ahmedwaleedmalik) [inline controllers/channel_controller.go:105]: Create a method to check `os.Getenv("ENABLE_UNARCHIVING_CHANNEL") == "true"`

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-32-e3f8150f`

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

**Comment by stakater-user** [comment]
@rehanjaved237 Image is available for testing. `docker pull stakater/slack-operator:SNAPSHOT-PR-32-551e1fff`

**Review by ahmedwaleedmalik — APPROVED** [review]
LGTM

**Pull request merged**